docs/cli/app-types.md

```diff
@@ -0,0 +1,141 @@
+# .NET Core CLI types of portability of applications
+
+## Introduction
+There are several ways to think about the "types" of applications you can build. Usually, the types describe 
+a certain execution model or are based on what the application can "do"; examples of these are "console application", 
+"web application" etc. All of these types of applications (and more) can be created with .NET Core, since it is 
+a general purpose development platform.
+
+However, given its unique nature, .NET Core also has another axis on which to observe the type of the application 
+and that is application's *portability*. Portability here means, essentially, where can you run your application and what are the 
+pre-requisites that you need to satisfy in order for your application to run on a given machine. This document 
+deals with this axis, portability, and outlines the two main types of portability that .NET Core enables. 
+
+There are two main types that we can observe: 
+
+1. Portable application
+    * As a sub-category of this we have the portable application with native dependencies
+2. Self-contained application
+
+## Portable application
+Fully portable application is an IL application that targets a shared component called **the shared runtime**. 
```

Comment:
Usually we say what acronyms mean on first mention. So here would be intermediate language (IL)